### PR TITLE
Enable OSTree FOTA for imx8mmevk.

### PR DESCRIPTION
**Pull request by petedyerarm**
Build with FOTA enabled and dynamically link with libcurl.
The install script, deploy_delta_update.sh, is placed into /wigwag/mbed on the
device and the fota-callback code is patched to use this location
instead of /opt/pelion.
/userdata is created in the rootfs as a mountpoint for the /userdata
partition.
/userdata is also used to store the FOTA artifacts, so that they persist
after a reboot.

### CODE REVIEW

File: recipes-connectivity/mbed-edge-core/mbed-edge-core.inc [hunk 1]
```diff
@@ -4,10 +4,10 @@ LICENSE="Apache-2.0"
 LIC_FILES_CHKSUM = "file://${WORKDIR}/git/LICENSE;md5=1dece7821bf3fd70fe1309eaa37d52a2"
 
 MBED_EDGE_CORE_CONFIG_TRACE_LEVEL ?= "DEBUG"
-MBED_EDGE_CORE_CONFIG_FIRMWARE_UPDATE ?= "OFF"
-MBED_EDGE_CORE_CONFIG_FOTA_ENABLE ?= "OFF"
-MBED_EDGE_CORE_CONFIG_FOTA_TRACE ?= "OFF"
-MBED_EDGE_CORE_CONFIG_CURL_DYNAMIC_LINK ?= "OFF"
```
Comment (x85446) [inline recipes-connectivity/mbed-edge-core/mbed-edge-core.inc:10]: @ygoyal18  Can you review this for changes to RPI?

**Comment by ygoyal18** [comment]
@esajaa @JanneKiiskila  Should we by default enable the FOTA FW and for RPi board add a step in our documentation to disable the FOTA before building? Apart from that everything looks good.

**Comment by JanneKiiskila** [comment]
Things should work straight out of box without any changes as much as possible. I think/guess this should make both of them work just fine?
- For RPI the UC_HUB is enabled by default (as it should).
- For FOTA_FW targets - this recipe fixes the configuration.
@petedyerarm @esajaa would you agree my guess is correct?

File: recipes-connectivity/mbed-edge-core/mbed-edge-core.inc [hunk 2]
```diff
@@ -4,10 +4,10 @@ LICENSE="Apache-2.0"
...
+MBED_EDGE_CORE_CONFIG_FIRMWARE_UPDATE ?= "ON"
+MBED_EDGE_CORE_CONFIG_FOTA_ENABLE ?= "ON"
+MBED_EDGE_CORE_CONFIG_FOTA_TRACE ?= "ON"
+MBED_EDGE_CORE_CONFIG_CURL_DYNAMIC_LINK ?= "ON"
```
Comment (ygoyal18) [inline recipes-connectivity/mbed-edge-core/mbed-edge-core.inc:10]: @JanneKiiskila Actually here we are by default enabling the FOTA FW library for all the platforms. So should we add conditions here such that we only enable UC_hub for RPi and FOTA FW for rest of the platforms?

Comment (JanneKiiskila) [reply recipes-connectivity/mbed-edge-core/mbed-edge-core.inc:10]: Ok, I'm not familiar (at all) with the Yocto/Bitbake stuff - but indeed, we should keep the default firmware update option on for each relevant target.
- UC_HUB for RPI3B
- FOTA_FW for i.MX8 and AVNET boards.

**Comment by ygoyal18** [comment]
@petedyerarm We should merge this PR - https://github.com/PelionIoT/meta-mbed-edge/pull/20 ahead of this one and then rebase to dev. This will help us manage different variants of Edge Core on different platforms.

**Review by ygoyal18 — APPROVED** [review]
LGTM

**Pull request merged**